jiant/tasks/tasks.py

```diff
@@ -3023,3 +3023,110 @@ def count_examples(self, splits=["train", "val", "test"]):
         for split in splits:
             st = self.get_split_text(split)
             self.example_counts[split] = len(st)
+
+
+@register_task("anli", rel_path="aNLI")
+class AlphaNLITask(MultipleChoiceTask):
+    """ Task class for Abductive Natural Language Inference. """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super().__init__(name, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+
+        self.scorer1 = CategoricalAccuracy()
+        self.scorers = [self.scorer1]
+        self.val_metric = "%s_accuracy" % name
+        self.val_metric_decreases = False
+        self.n_choices = 2
+
+    def load_data(self):
+        """ Process the dataset located at path.  """
+
+        def _load_split(inputs_file, labels_file):
+            obs1, hyp1, hyp2, obs2 = [], [], [], []
+            with open(inputs_file, encoding="utf-8") as f:
+                for line in f:
+                    row = json.loads(line)
+                    obs1.append(
+                        tokenize_and_truncate(self._tokenizer_name, row["obs1"], self.max_seq_len)
+                    )
+                    hyp1.append(
+                        tokenize_and_truncate(self._tokenizer_name, row["hyp1"], self.max_seq_len)
+                    )
+                    hyp2.append(
+                        tokenize_and_truncate(self._tokenizer_name, row["hyp2"], self.max_seq_len)
+                    )
+                    obs2.append(
+                        tokenize_and_truncate(self._tokenizer_name, row["obs2"], self.max_seq_len)
+                    )
+            with open(labels_file) as f:
+                labels = [int(i) - 1 for i in f.read().split()]  # -1 to get {0, 1} labels
+            return [obs1, hyp1, hyp2, obs2, labels]
+
+        self.train_data_text = _load_split(
+            inputs_file=os.path.join(self.path, "train.jsonl"),
+            labels_file=os.path.join(self.path, "train-labels.lst"),
+        )
+        self.val_data_text = _load_split(
+            inputs_file=os.path.join(self.path, "dev.jsonl"),
+            labels_file=os.path.join(self.path, "dev-labels.lst"),
+        )
+        self.test_data_text = [], [], [], [], []
```

Comment:
Double check that we don't assume the existence of a test set anywhere. Could this do bad things if we do a standard eval with multiple different target tasks?

Alternative would be to set test = dev, and just add a warning here.